src/mcpm/router/router.py

```diff
@@ -38,19 +38,28 @@ class MCPRouter:
     exposes them as a single SSE server.
     """
 
-    def __init__(self, reload_server: bool = False) -> None:
-        """Initialize the router."""
+    def __init__(self, reload_server: bool = False, profile_path: str | None = None, strict: bool = False) -> None:
+        """
+        Initialize the router.
+
+        :param reload_server: Whether to reload the server when the config changes
+        :param profile_path: Path to the profile file
+        :param strict: Whether to use strict mode for duplicated tool name.
+                       If True, raise error when duplicated tool name is found else auto resolve by adding server name prefix
+        """
         self.server_sessions: t.Dict[str, ServerConnection] = {}
         self.capabilities_mapping: t.Dict[str, t.Dict[str, t.Any]] = defaultdict(dict)
-        self.tools_mapping: t.Dict[str, t.Dict[str, t.Any]] = {}
-        self.prompts_mapping: t.Dict[str, t.Dict[str, t.Any]] = {}
-        self.resources_mapping: t.Dict[str, t.Dict[str, t.Any]] = {}
-        self.resources_templates_mapping: t.Dict[str, t.Dict[str, t.Any]] = {}
+        self.capabilities_to_server_id: t.Dict[str, t.Dict[str, t.Any]] = defaultdict(dict)
```

Comment:
rename to `capabilities_to_server_id` to store not only tools but all capabilities